.pre-commit-config.yaml

```diff
@@ -15,59 +15,24 @@ repos:
       - id: requirements-txt-fixer
       - id: trailing-whitespace
 
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v3.16.0
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.5.0
     hooks:
-      - id: pyupgrade
-        args:
-          - --py39-plus
-
-  - repo: https://github.com/pycqa/isort
-    rev: 5.13.2
-    hooks:
-      - id: isort
-        name: isort
+      - id: ruff
+        args: [--fix]
 
```

Comment:
How did you decide to choose `black` over `ruff format`?

Reply:
I didn't decide this if we are fully moving to ruff then I am going to remove the black format completely.

Reply:
@fkiraly @benHeid do you have any thoughts on keeping `black` or switching to `ruff` completely?

I've no strong opinion, and if `ruff` is faster with (mostly) same styling, I'll favour the switch slightly.